src/arguments.jl

```diff
@@ -0,0 +1,36 @@
+"""
+    Arguments{names,T}
+
+    Arguments(; kw...)
+    Arguments(nt::NamedTuple)
+
+A wrapper for a `NamedTuple`.
+"""
+struct Arguments{names,T}
+    nt::NamedTuple{names,T}
+end
+
+Arguments(; kw...) = Arguments(values(kw))
+Arguments{names}(vals) where {names} = Arguments(NamedTuple{names}(vals))
+
+nt(a::Arguments) = getfield(a, :nt)
+
+for f in (:length, :firstindex, :lastindex, :isempty, :keys, :values)
```

Comment:
Do we need to define `propertynames` as well?

Reply:
doesn't cost much